### PR TITLE
[C-API] Fix the Tizen ACR issue

### PR DESCRIPTION
**Pull request by again4you**
To comply with Tizen ACR process, this set of patches newly add some sample code and updates comments for doxygen. Detailed items are as below.

* Add PERMISSION_DENIED state for privilege issue
* Apply the review guide of C-API
* Describe the pipeline state as image instead of text art
* Describe pre & post state of the pipeline
* Add sample code for ml_pipeline_switch_get_pad_list()

Signed-off-by: Sangjung Woo <sangjung.woo@samsung.com>

**Self evaluation:**
1. Build test: [* ]Passed [ ]Failed []Skipped
2. Run test: [* ]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: api/capi/include/nnstreamer-capi-private.h [hunk 1]
```diff
@@ -36,8 +36,9 @@
 #define ML_INF_FEATURE_PATH "tizen.org/feature/machine_learning.inference"
 
 #define check_feature_state() \
-  if (ML_ERROR_NONE != ml_get_feature_enabled()) \
-    return ML_ERROR_NOT_SUPPORTED;
+  int feature_ret = ml_get_feature_enabled(); \
+  if (ML_ERROR_NONE != feature_ret) \
+    return feature_ret;
```
Comment (myungjoo) [inline api/capi/include/nnstreamer-capi-private.h]:
```
do { \
  int feature_ret = ml_get_feature_enabled(); \
  if (ML_ERROR_NONE != feature_ret) \
    return feature_ret; \
} while (0);
```

File: api/capi/include/nnstreamer.h [hunk 2]
```diff
@@ -153,6 +153,7 @@ typedef enum {
   ML_ERROR_UNKNOWN              = TIZEN_ERROR_UNKNOWN,  /**< Unknown error */
   ML_ERROR_TIMED_OUT            = TIZEN_ERROR_TIMED_OUT,  /**< Time out */
   ML_ERROR_NOT_SUPPORTED        = TIZEN_ERROR_NOT_SUPPORTED, /**< The feature is not supported */
+  ML_ERROR_PERMISSION_DENIED    = TIZEN_ERROR_PERMISSION_DENIED, /**< Permission denied */
```
Comment (myungjoo) [inline api/capi/include/nnstreamer.h]: Add this at ```modify_nnstreamer_h_for_nontizen.sh```.

Comment (again4you) [reply api/capi/include/nnstreamer.h]: Right. I will.

File: api/capi/doc/nnstreamer_doc.h [hunk 3]
```diff
@@ -43,6 +43,9 @@
  *
  *  Note that this function set is supposed to be thread-safe.
  *
+ * @subsection CAPI_MEDIA_RECORDER_MODULE_PIPELINE_STATE_DIAGRAM Pipeline State Diagram
```
Comment (jaeyun-jung) [inline api/capi/doc/nnstreamer_doc.h]: Question : ```CAPI_MEDIA_RECORDER_MODULE``` is it correct?

File: api/capi/include/nnstreamer.h [hunk 4]
```diff
@@ -444,13 +446,53 @@ int ml_pipeline_switch_select (ml_pipeline_switch_h switch_handle, const char *p
 /**
  * @brief Gets the pad names of a switch.
  * @since_tizen 5.5
+ * @remarks If the function succeeds, @a list and its contents be released using g_free(). Refer the below sample code.
```
Comment (jaeyun-jung) [inline api/capi/include/nnstreamer.h]: maybe ```should be released```

File: api/capi/include/nnstreamer-single.h [hunk 5]
```diff
@@ -74,6 +74,7 @@ typedef void *ml_single_h;
  * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
  * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
  * @retval #ML_ERROR_STREAMS_PIPE Failed to start the pipeline.
+ * @retval #ML_ERROR_PERMISSION_DENIED The application does not have the privilege to access to the media storage or external storage.
```
Comment (jaeyun-jung) [inline api/capi/include/nnstreamer-single.h]: If the function check_feature_state() returns ML_ERROR_PERMISSION_DENIED, then do we add this error code in all C-API functions?

File: api/capi/include/nnstreamer-capi-private.h (hunk 1 above)
Comment (again4you) [reply api/capi/include/nnstreamer-capi-private.h]: I will.

File: api/capi/doc/nnstreamer_doc.h (hunk 3 above)
Comment (again4you) [reply api/capi/doc/nnstreamer_doc.h]: Sorry. I made a mistake. `CAPI_ML_NNSTREAMER_PIPELINE_STATE_DIAGRAM` is right.

File: api/capi/include/nnstreamer.h (hunk 4 above)
Comment (again4you) [reply api/capi/include/nnstreamer.h]: Right.

**Comment by again4you** [comment]
### Submit v2
* Fix the item that reviewer mentioned.

File: api/capi/include/nnstreamer.h [hunk 6]
```diff
@@ -167,24 +168,8 @@ typedef enum {
...
- *          [  READY  ]                                            |
- *               | "ml_pipeline_construct' completes              | "ml_pipeline_destroy"
- *               V                                                 |
- *          [         ] ------------------------------------------>|
- *          [  PAUSED ] <-------------------+                      |
- *               | "ml_pipeline_start"     | "ml_pipeline_stop"  |
- *               V                          |                      |
- *          [ PLAYING ] --------------------+----------------------+
- *
+ * @details The pipeline state is described on @ref CAPI_MEDIA_RECORDER_MODULE_PIPELINE_STATE_DIAGRAM.
```
Comment (jaeyun-jung) [inline api/capi/include/nnstreamer.h]: plz change to CAPI_ML_NNSTREAMER_PIPELINE_STATE_DIAGRAM

**Comment by again4you** [comment]
### Submit v3
* Update the sample code for `ml_pipeline_switch_get_pad_list()`
* Fix the omitted item

**Pull request merged**